### ISSUE TITLE
[Error] Exported running app icon not showing correctly

### ISSUE THREAD

**Issue by cdock1029** [89luca89/distrobox#1306]
**Describe the bug**

On fresh install of openSUSE Aeon running Gnome 46, distrobox exported apps show correct app icon in the dash and the app grid, but show generic placeholder icon for the running app. Also, the running app displays separate from the icon pinned in dash. Observed this with vscode and qtcreator exported from tumbleweed distrobox.

![Screenshot from 2024-03-23 16-51-22](https://github.com/89luca89/distrobox/assets/802586/1d51ade1-58b7-49b2-86c5-b861cf5baca8)

**To Reproduce**

Install openSUSE Aeon running Gnome 46

`distrobox enter`
`sudo zypper in opi`
`opi vscode`
`distrobox-export --app code`

Log out, log in. See correct icon displayed in app grid and after pinning to dash.
Run app, see generic icon in dash tray.

Same after installing qtcreator.

**Expected behavior**

The running app would associate with the pinned dash icon or display separate in the dash but in either case use the correct app icon.

**Logs**

<details>
  <summary>podman logs --latest</summary>

```
+ stat /run/host/etc/shadow
++ stat -c %u /run/host/etc/shadow
+ '[' 65534 = 0 ']'
+ '[' -n '' ']'
+ printf 'distrobox: Installing basic packages...\n'
distrobox: Installing basic packages...
++ basename bash
+ shell_pkg=bash
+ '[' bash = ash ']'
+ missing_packages=0
+ dependencies='
	bc
	bzip2
	chpasswd
	curl
	diff
	find
	findmnt
	gpg
	hostname
	less
	lsof
	man
	mount
	passwd
	pigz
	pinentry
	ping
	ps
	rsync
	script
	ssh
	sudo
	time
	tree
	umount
	unzip
	useradd
	wc
	wget
	xauth
	zip
	bash
'
+ for dep in ${dependencies}
+ command -v bc
+ for dep in ${dependencies}
+ command -v bzip2
+ for dep in ${dependencies}
+ command -v chpasswd
+ for dep in ${dependencies}
+ command -v curl
+ for dep in ${dependencies}
+ command -v diff
+ for dep in ${dependencies}
+ command -v find
+ for dep in ${dependencies}
+ command -v findmnt
+ for dep in ${dependencies}
+ command -v gpg
+ for dep in ${dependencies}
+ command -v hostname
+ for dep in ${dependencies}
+ command -v less
+ for dep in ${dependencies}
+ command -v lsof
+ for dep in ${dependencies}
+ command -v man
+ for dep in ${dependencies}
+ command -v mount
+ for dep in ${dependencies}
+ command -v passwd
+ for dep in ${dependencies}
+ command -v pigz
+ for dep in ${dependencies}
+ command -v pinentry
+ for dep in ${dependencies}
+ command -v ping
+ for dep in ${dependencies}
+ command -v ps
+ for dep in ${dependencies}
+ command -v rsync
+ for dep in ${dependencies}
+ command -v script
+ for dep in ${dependencies}
+ command -v ssh
+ for dep in ${dependencies}
+ command -v sudo
+ for dep in ${dependencies}
+ command -v time
+ for dep in ${dependencies}
+ command -v tree
+ for dep in ${dependencies}
+ command -v umount
+ for dep in ${dependencies}
+ command -v unzip
+ for dep in ${dependencies}
+ command -v useradd
+ for dep in ${dependencies}
+ command -v wc
+ for dep in ${dependencies}
+ command -v wget
+ for dep in ${dependencies}
+ command -v xauth
+ for dep in ${dependencies}
+ command -v zip
+ for dep in ${dependencies}
+ command -v bash
+ PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/bin:/sbin:/usr/bin:/usr/sbin
+ '[' 0 -ne 0 ']'
+ '[' 0 -ne 0 ']'
+ '[' -n '' ']'
++ command -v bash
+ SHELL=/usr/bin/bash
+ /usr/bin/distrobox-host-exec -Y test
+ :
+ command -v xdg-open
+ mkdir -p /usr/local/bin/
+ ln -sf /usr/bin/distrobox-host-exec /usr/local/bin/xdg-open
+ su --help
+ grep -q pty
+ printf 'distrobox: Setting up devpts mounts...\n'
distrobox: Setting up devpts mounts...
+ grep -q tty /etc/group
+ printf %s tty:x:5:
+ mount -t devpts devpts -o noexec,nosuid,newinstance,ptmxmode=0666,mode=0620,gid=tty /dev/pts/
+ mount --bind /dev/pts/ptmx /dev/ptmx
+ mount --make-rshared /
+ printf 'distrobox: Setting up read-only mounts...\n'
distrobox: Setting up read-only mounts...
+ HOST_MOUNTS_RO='
	/etc/localtime
	/var/lib/systemd/coredump
	/var/log/journal'
+ for host_mount_ro in ${HOST_MOUNTS_RO}
++ get_locked_mount_flags /run/host/etc/localtime
++ source=/run/host/etc/localtime
++ prev=
++ locked_flags=
++ ls /run/host/etc/localtime
++ true
+++ findmnt --noheadings --output OPTIONS --target /run/host/etc/localtime
++ flags=ro,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=270,subvol=/@/.snapshots/4/snapshot
++ '[' -n ro,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=270,subvol=/@/.snapshots/4/snapshot ']'
++ break
++ for flag in nodev noexec nosuid
++ printf %s ro,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=270,subvol=/@/.snapshots/4/snapshot
++ grep -q nodev
++ for flag in nodev noexec nosuid
++ printf %s ro,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=270,subvol=/@/.snapshots/4/snapshot
++ grep -q noexec
++ for flag in nodev noexec nosuid
++ printf %s ro,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=270,subvol=/@/.snapshots/4/snapshot
++ grep -q nosuid
++ printf %s ''
+ locked_flags=
+ mount_bind /run/host/etc/localtime /etc/localtime ro
+ source_dir=/run/host/etc/localtime
+ target_dir=/etc/localtime
+ mount_flags=
+ '[' 3 -gt 2 ']'
+ mount_flags=ro
+ '[' -L /run/host/etc/localtime ']'
++ init_readlink /run/host/etc/localtime
++ grep -Eo '\->.*'
++ ls -l /run/host/etc/localtime
++ cut '-d ' -f2-
++ sed 's|\.\./|/|g'
+ source_dir=/usr/share/zoneinfo/America/New_York
+ printf %s /usr/share/zoneinfo/America/New_York
+ grep -q /run/host
+ source_dir=/run/host/usr/share/zoneinfo/America/New_York
+ '[' '!' -d /run/host/usr/share/zoneinfo/America/New_York ']'
+ '[' '!' -f /run/host/usr/share/zoneinfo/America/New_York ']'
+ '[' -e /etc/localtime ']'
+ findmnt /etc/localtime
+ '[' -L /etc/localtime ']'
+ rm -f /etc/localtime
+ '[' -d /run/host/usr/share/zoneinfo/America/New_York ']'
+ '[' -f /run/host/usr/share/zoneinfo/America/New_York ']'
++ dirname /etc/localtime
+ '[' '!' -d /etc ']'
+ touch /etc/localtime
+ '[' ro = '' ']'
+ mount --rbind -o ro /run/host/usr/share/zoneinfo/America/New_York /etc/localtime
+ return 0
+ for host_mount_ro in ${HOST_MOUNTS_RO}
++ get_locked_mount_flags /run/host/var/lib/systemd/coredump
++ source=/run/host/var/lib/systemd/coredump
++ prev=
++ locked_flags=
++ ls /run/host/var/lib/systemd/coredump
++ true
+++ findmnt --noheadings --output OPTIONS --target /run/host/var/lib/systemd/coredump
++ flags=rw,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=257,subvol=/@/var
++ '[' -n rw,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=257,subvol=/@/var ']'
++ break
++ for flag in nodev noexec nosuid
++ printf %s rw,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=257,subvol=/@/var
++ grep -q nodev
++ for flag in nodev noexec nosuid
++ printf %s rw,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=257,subvol=/@/var
++ grep -q noexec
++ for flag in nodev noexec nosuid
++ printf %s rw,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=257,subvol=/@/var
++ grep -q nosuid
++ printf %s ''
+ locked_flags=
+ mount_bind /run/host/var/lib/systemd/coredump /var/lib/systemd/coredump ro
+ source_dir=/run/host/var/lib/systemd/coredump
+ target_dir=/var/lib/systemd/coredump
+ mount_flags=
+ '[' 3 -gt 2 ']'
+ mount_flags=ro
+ '[' -L /run/host/var/lib/systemd/coredump ']'
+ '[' '!' -d /run/host/var/lib/systemd/coredump ']'
+ '[' -e /var/lib/systemd/coredump ']'
+ '[' -L /var/lib/systemd/coredump ']'
+ '[' -d /run/host/var/lib/systemd/coredump ']'
+ mkdir -p /var/lib/systemd/coredump
+ '[' ro = '' ']'
+ mount --rbind -o ro /run/host/var/lib/systemd/coredump /var/lib/systemd/coredump
+ return 0
+ for host_mount_ro in ${HOST_MOUNTS_RO}
++ get_locked_mount_flags /run/host/var/log/journal
++ source=/run/host/var/log/journal
++ prev=
++ locked_flags=
++ ls /run/host/var/log/journal
++ true
+++ findmnt --noheadings --output OPTIONS --target /run/host/var/log/journal
++ flags=rw,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=257,subvol=/@/var
++ '[' -n rw,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=257,subvol=/@/var ']'
++ break
++ for flag in nodev noexec nosuid
++ printf %s rw,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=257,subvol=/@/var
++ grep -q nodev
++ for flag in nodev noexec nosuid
++ printf %s rw,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=257,subvol=/@/var
++ grep -q noexec
++ for flag in nodev noexec nosuid
++ printf %s rw,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=257,subvol=/@/var
++ grep -q nosuid
++ printf %s ''
+ locked_flags=
+ mount_bind /run/host/var/log/journal /var/log/journal ro
+ source_dir=/run/host/var/log/journal
+ target_dir=/var/log/journal
+ mount_flags=
+ '[' 3 -gt 2 ']'
+ mount_flags=ro
+ '[' -L /run/host/var/log/journal ']'
+ '[' '!' -d /run/host/var/log/journal ']'
+ '[' -e /var/log/journal ']'
+ findmnt /var/log/journal
+ umount /var/log/journal
+ '[' -L /var/log/journal ']'
+ '[' -d /run/host/var/log/journal ']'
+ mkdir -p /var/log/journal
+ '[' ro = '' ']'
+ mount --rbind -o ro /run/host/var/log/journal /var/log/journal
+ return 0
+ printf 'distrobox: Setting up read-write mounts...\n'
distrobox: Setting up read-write mounts...
+ '[' -e /var/home/cdock ']'
+ HOST_MOUNTS='
	/etc/host.conf
	/etc/machine-id
	/media
	/mnt
	/run/libvirt
	/run/media
	/run/netconfig/
	/run/systemd/journal
	/run/systemd/resolve/
	/run/systemd/seats
	/run/systemd/sessions
	/run/systemd/users
	/run/udev
	/var/lib/libvirt
	/var/mnt'
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/etc/host.conf /etc/host.conf
+ source_dir=/run/host/etc/host.conf
+ target_dir=/etc/host.conf
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/etc/host.conf ']'
+ '[' '!' -d /run/host/etc/host.conf ']'
+ '[' '!' -f /run/host/etc/host.conf ']'
+ '[' -e /etc/host.conf ']'
+ findmnt /etc/host.conf
+ '[' -L /etc/host.conf ']'
+ '[' -d /run/host/etc/host.conf ']'
+ '[' -f /run/host/etc/host.conf ']'
++ dirname /etc/host.conf
+ '[' '!' -d /etc ']'
+ touch /etc/host.conf
+ '[' '' = '' ']'
+ mount_flags=rslave
+ mount --rbind -o rslave /run/host/etc/host.conf /etc/host.conf
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/etc/machine-id /etc/machine-id
+ source_dir=/run/host/etc/machine-id
+ target_dir=/etc/machine-id
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/etc/machine-id ']'
+ '[' '!' -d /run/host/etc/machine-id ']'
+ '[' '!' -f /run/host/etc/machine-id ']'
+ '[' -e /etc/machine-id ']'
+ findmnt /etc/machine-id
+ '[' -L /etc/machine-id ']'
+ '[' -d /run/host/etc/machine-id ']'
+ '[' -f /run/host/etc/machine-id ']'
++ dirname /etc/machine-id
+ '[' '!' -d /etc ']'
+ touch /etc/machine-id
+ '[' '' = '' ']'
+ mount_flags=rslave
+ mount --rbind -o rslave /run/host/etc/machine-id /etc/machine-id
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/media /media
+ source_dir=/run/host/media
+ target_dir=/media
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/media ']'
+ '[' '!' -d /run/host/media ']'
+ '[' '!' -f /run/host/media ']'
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/mnt /mnt
+ source_dir=/run/host/mnt
+ target_dir=/mnt
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/mnt ']'
+ '[' '!' -d /run/host/mnt ']'
+ '[' -e /mnt ']'
+ findmnt /mnt
+ '[' -L /mnt ']'
+ '[' -d /run/host/mnt ']'
+ mkdir -p /mnt
+ '[' '' = '' ']'
+ mount_flags=rslave
+ mount --rbind -o rslave /run/host/mnt /mnt
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/run/libvirt /run/libvirt
+ source_dir=/run/host/run/libvirt
+ target_dir=/run/libvirt
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/run/libvirt ']'
+ '[' '!' -d /run/host/run/libvirt ']'
+ '[' '!' -f /run/host/run/libvirt ']'
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/run/media /run/media
+ source_dir=/run/host/run/media
+ target_dir=/run/media
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/run/media ']'
+ '[' '!' -d /run/host/run/media ']'
+ '[' -e /run/media ']'
+ '[' -L /run/media ']'
+ '[' -d /run/host/run/media ']'
+ mkdir -p /run/media
+ '[' '' = '' ']'
+ mount_flags=rslave
+ mount --rbind -o rslave /run/host/run/media /run/media
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/run/netconfig/ /run/netconfig/
+ source_dir=/run/host/run/netconfig/
+ target_dir=/run/netconfig/
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/run/netconfig/ ']'
+ '[' '!' -d /run/host/run/netconfig/ ']'
+ '[' '!' -f /run/host/run/netconfig/ ']'
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/run/systemd/journal /run/systemd/journal
+ source_dir=/run/host/run/systemd/journal
+ target_dir=/run/systemd/journal
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/run/systemd/journal ']'
+ '[' '!' -d /run/host/run/systemd/journal ']'
+ '[' -e /run/systemd/journal ']'
+ '[' -L /run/systemd/journal ']'
+ '[' -d /run/host/run/systemd/journal ']'
+ mkdir -p /run/systemd/journal
+ '[' '' = '' ']'
+ mount_flags=rslave
+ mount --rbind -o rslave /run/host/run/systemd/journal /run/systemd/journal
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/run/systemd/resolve/ /run/systemd/resolve/
+ source_dir=/run/host/run/systemd/resolve/
+ target_dir=/run/systemd/resolve/
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/run/systemd/resolve/ ']'
+ '[' '!' -d /run/host/run/systemd/resolve/ ']'
+ '[' '!' -f /run/host/run/systemd/resolve/ ']'
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/run/systemd/seats /run/systemd/seats
+ source_dir=/run/host/run/systemd/seats
+ target_dir=/run/systemd/seats
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/run/systemd/seats ']'
+ '[' '!' -d /run/host/run/systemd/seats ']'
+ '[' -e /run/systemd/seats ']'
+ '[' -L /run/systemd/seats ']'
+ '[' -d /run/host/run/systemd/seats ']'
+ mkdir -p /run/systemd/seats
+ '[' '' = '' ']'
+ mount_flags=rslave
+ mount --rbind -o rslave /run/host/run/systemd/seats /run/systemd/seats
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/run/systemd/sessions /run/systemd/sessions
+ source_dir=/run/host/run/systemd/sessions
+ target_dir=/run/systemd/sessions
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/run/systemd/sessions ']'
+ '[' '!' -d /run/host/run/systemd/sessions ']'
+ '[' -e /run/systemd/sessions ']'
+ '[' -L /run/systemd/sessions ']'
+ '[' -d /run/host/run/systemd/sessions ']'
+ mkdir -p /run/systemd/sessions
+ '[' '' = '' ']'
+ mount_flags=rslave
+ mount --rbind -o rslave /run/host/run/systemd/sessions /run/systemd/sessions
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/run/systemd/users /run/systemd/users
+ source_dir=/run/host/run/systemd/users
+ target_dir=/run/systemd/users
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/run/systemd/users ']'
+ '[' '!' -d /run/host/run/systemd/users ']'
+ '[' -e /run/systemd/users ']'
+ '[' -L /run/systemd/users ']'
+ '[' -d /run/host/run/systemd/users ']'
+ mkdir -p /run/systemd/users
+ '[' '' = '' ']'
+ mount_flags=rslave
+ mount --rbind -o rslave /run/host/run/systemd/users /run/systemd/users
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/run/udev /run/udev
+ source_dir=/run/host/run/udev
+ target_dir=/run/udev
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/run/udev ']'
+ '[' '!' -d /run/host/run/udev ']'
+ '[' -e /run/udev ']'
+ '[' -L /run/udev ']'
+ '[' -d /run/host/run/udev ']'
+ mkdir -p /run/udev
+ '[' '' = '' ']'
+ mount_flags=rslave
+ mount --rbind -o rslave /run/host/run/udev /run/udev
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/var/lib/libvirt /var/lib/libvirt
+ source_dir=/run/host/var/lib/libvirt
+ target_dir=/var/lib/libvirt
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/var/lib/libvirt ']'
+ '[' '!' -d /run/host/var/lib/libvirt ']'
+ '[' '!' -f /run/host/var/lib/libvirt ']'
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/var/mnt /var/mnt
+ source_dir=/run/host/var/mnt
+ target_dir=/var/mnt
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/var/mnt ']'
+ '[' '!' -d /run/host/var/mnt ']'
+ '[' '!' -f /run/host/var/mnt ']'
+ return 0
+ printf 'distrobox: Setting up host'\''s sockets integration...\n'
distrobox: Setting up host's sockets integration...
++ find /run/host/run -path /run/host/run/media -prune -o -path /run/host/run/timeshift -prune -o -name user -prune -o -name bees -prune -o -name nscd -prune -o -name system_bus_socket -prune -o -type s -print
++ :
+ host_sockets='/run/host/run/systemd/inaccessible/sock
/run/host/run/systemd/notify
/run/host/run/systemd/userdb/io.systemd.DynamicUser
/run/host/run/systemd/io.systemd.ManagedOOM
/run/host/run/systemd/journal/dev-log
/run/host/run/systemd/journal/socket
/run/host/run/systemd/journal/stdout
/run/host/run/systemd/journal/io.systemd.journal
/run/host/run/systemd/private
/run/host/run/systemd/coredump
/run/host/run/systemd/io.systemd.sysext
/run/host/run/udev/control
/run/host/run/cups/cups.sock
/run/host/run/avahi-daemon/socket
/run/host/run/irqbalance/irqbalance1065.sock'
+ for host_socket in ${host_sockets}
++ printf %s /run/host/run/systemd/inaccessible/sock
++ sed 's|/run/host||g'
+ container_socket=/run/systemd/inaccessible/sock
+ '[' '!' -S /run/systemd/inaccessible/sock ']'
+ '[' '!' -L /run/systemd/inaccessible/sock ']'
+ rm -f /run/systemd/inaccessible/sock
++ dirname /run/systemd/inaccessible/sock
+ mkdir -p /run/systemd/inaccessible
+ ln -s /run/host/run/systemd/inaccessible/sock /run/systemd/inaccessible/sock
+ for host_socket in ${host_sockets}
++ printf %s /run/host/run/systemd/notify
++ sed 's|/run/host||g'
+ container_socket=/run/systemd/notify
+ '[' '!' -S /run/systemd/notify ']'
+ '[' '!' -L /run/systemd/notify ']'
+ rm -f /run/systemd/notify
++ dirname /run/systemd/notify
+ mkdir -p /run/systemd
+ ln -s /run/host/run/systemd/notify /run/systemd/notify
+ for host_socket in ${host_sockets}
++ printf %s /run/host/run/systemd/userdb/io.systemd.DynamicUser
++ sed 's|/run/host||g'
+ container_socket=/run/systemd/userdb/io.systemd.DynamicUser
+ '[' '!' -S /run/systemd/userdb/io.systemd.DynamicUser ']'
+ '[' '!' -L /run/systemd/userdb/io.systemd.DynamicUser ']'
+ rm -f /run/systemd/userdb/io.systemd.DynamicUser
++ dirname /run/systemd/userdb/io.systemd.DynamicUser
+ mkdir -p /run/systemd/userdb
+ ln -s /run/host/run/systemd/userdb/io.systemd.DynamicUser /run/systemd/userdb/io.systemd.DynamicUser
+ for host_socket in ${host_sockets}
++ printf %s /run/host/run/systemd/io.systemd.ManagedOOM
++ sed 's|/run/host||g'
+ container_socket=/run/systemd/io.systemd.ManagedOOM
+ '[' '!' -S /run/systemd/io.systemd.ManagedOOM ']'
+ '[' '!' -L /run/systemd/io.systemd.ManagedOOM ']'
+ rm -f /run/systemd/io.systemd.ManagedOOM
++ dirname /run/systemd/io.systemd.ManagedOOM
+ mkdir -p /run/systemd
+ ln -s /run/host/run/systemd/io.systemd.ManagedOOM /run/systemd/io.systemd.ManagedOOM
+ for host_socket in ${host_sockets}
++ printf %s /run/host/run/systemd/journal/dev-log
++ sed 's|/run/host||g'
+ container_socket=/run/systemd/journal/dev-log
+ '[' '!' -S /run/systemd/journal/dev-log ']'
+ for host_socket in ${host_sockets}
++ sed 's|/run/host||g'
++ printf %s /run/host/run/systemd/journal/socket
+ container_socket=/run/systemd/journal/socket
+ '[' '!' -S /run/systemd/journal/socket ']'
+ for host_socket in ${host_sockets}
++ printf %s /run/host/run/systemd/journal/stdout
++ sed 's|/run/host||g'
+ container_socket=/run/systemd/journal/stdout
+ '[' '!' -S /run/systemd/journal/stdout ']'
+ for host_socket in ${host_sockets}
++ printf %s /run/host/run/systemd/journal/io.systemd.journal
++ sed 's|/run/host||g'
+ container_socket=/run/systemd/journal/io.systemd.journal
+ '[' '!' -S /run/systemd/journal/io.systemd.journal ']'
+ for host_socket in ${host_sockets}
++ printf %s /run/host/run/systemd/private
++ sed 's|/run/host||g'
+ container_socket=/run/systemd/private
+ '[' '!' -S /run/systemd/private ']'
+ '[' '!' -L /run/systemd/private ']'
+ rm -f /run/systemd/private
++ dirname /run/systemd/private
+ mkdir -p /run/systemd
+ ln -s /run/host/run/systemd/private /run/systemd/private
+ for host_socket in ${host_sockets}
++ printf %s /run/host/run/systemd/coredump
++ sed 's|/run/host||g'
+ container_socket=/run/systemd/coredump
+ '[' '!' -S /run/systemd/coredump ']'
+ '[' '!' -L /run/systemd/coredump ']'
+ rm -f /run/systemd/coredump
++ dirname /run/systemd/coredump
+ mkdir -p /run/systemd
+ ln -s /run/host/run/systemd/coredump /run/systemd/coredump
+ for host_socket in ${host_sockets}
++ printf %s /run/host/run/systemd/io.systemd.sysext
++ sed 's|/run/host||g'
+ container_socket=/run/systemd/io.systemd.sysext
+ '[' '!' -S /run/systemd/io.systemd.sysext ']'
+ '[' '!' -L /run/systemd/io.systemd.sysext ']'
+ rm -f /run/systemd/io.systemd.sysext
++ dirname /run/systemd/io.systemd.sysext
+ mkdir -p /run/systemd
+ ln -s /run/host/run/systemd/io.systemd.sysext /run/systemd/io.systemd.sysext
+ for host_socket in ${host_sockets}
++ printf %s /run/host/run/udev/control
++ sed 's|/run/host||g'
+ container_socket=/run/udev/control
+ '[' '!' -S /run/udev/control ']'
+ for host_socket in ${host_sockets}
++ printf %s /run/host/run/cups/cups.sock
++ sed 's|/run/host||g'
+ container_socket=/run/cups/cups.sock
+ '[' '!' -S /run/cups/cups.sock ']'
+ '[' '!' -L /run/cups/cups.sock ']'
+ rm -f /run/cups/cups.sock
++ dirname /run/cups/cups.sock
+ mkdir -p /run/cups
+ ln -s /run/host/run/cups/cups.sock /run/cups/cups.sock
+ for host_socket in ${host_sockets}
++ printf %s /run/host/run/avahi-daemon/socket
++ sed 's|/run/host||g'
+ container_socket=/run/avahi-daemon/socket
+ '[' '!' -S /run/avahi-daemon/socket ']'
+ '[' '!' -L /run/avahi-daemon/socket ']'
+ rm -f /run/avahi-daemon/socket
++ dirname /run/avahi-daemon/socket
+ mkdir -p /run/avahi-daemon
+ ln -s /run/host/run/avahi-daemon/socket /run/avahi-daemon/socket
+ for host_socket in ${host_sockets}
++ printf %s /run/host/run/irqbalance/irqbalance1065.sock
++ sed 's|/run/host||g'
+ container_socket=/run/irqbalance/irqbalance1065.sock
+ '[' '!' -S /run/irqbalance/irqbalance1065.sock ']'
+ '[' '!' -L /run/irqbalance/irqbalance1065.sock ']'
+ rm -f /run/irqbalance/irqbalance1065.sock
++ dirname /run/irqbalance/irqbalance1065.sock
+ mkdir -p /run/irqbalance
+ ln -s /run/host/run/irqbalance/irqbalance1065.sock /run/irqbalance/irqbalance1065.sock
+ '[' 0 -eq 1 ']'
+ printf 'distrobox: Integrating host'\''s themes, icons, fonts...\n'
distrobox: Integrating host's themes, icons, fonts...
+ mount_bind /run/host/usr/share/themes /usr/local/share/themes
+ source_dir=/run/host/usr/share/themes
+ target_dir=/usr/local/share/themes
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/usr/share/themes ']'
+ '[' '!' -d /run/host/usr/share/themes ']'
+ '[' -e /usr/local/share/themes ']'
+ '[' -L /usr/local/share/themes ']'
+ '[' -d /run/host/usr/share/themes ']'
+ mkdir -p /usr/local/share/themes
+ '[' '' = '' ']'
+ mount_flags=rslave
+ mount --rbind -o rslave /run/host/usr/share/themes /usr/local/share/themes
+ return 0
+ mount_bind /run/host/usr/share/icons /usr/local/share/icons
+ source_dir=/run/host/usr/share/icons
+ target_dir=/usr/local/share/icons
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/usr/share/icons ']'
+ '[' '!' -d /run/host/usr/share/icons ']'
+ '[' -e /usr/local/share/icons ']'
+ '[' -L /usr/local/share/icons ']'
+ '[' -d /run/host/usr/share/icons ']'
+ mkdir -p /usr/local/share/icons
+ '[' '' = '' ']'
+ mount_flags=rslave
+ mount --rbind -o rslave /run/host/usr/share/icons /usr/local/share/icons
+ return 0
+ mount_bind /run/host/usr/share/fonts /usr/local/share/fonts
+ source_dir=/run/host/usr/share/fonts
+ target_dir=/usr/local/share/fonts
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/usr/share/fonts ']'
+ '[' '!' -d /run/host/usr/share/fonts ']'
+ '[' -e /usr/local/share/fonts ']'
+ '[' -L /usr/local/share/fonts ']'
+ '[' -d /run/host/usr/share/fonts ']'
+ mkdir -p /usr/local/share/fonts
+ '[' '' = '' ']'
+ mount_flags=rslave
+ mount --rbind -o rslave /run/host/usr/share/fonts /usr/local/share/fonts
+ return 0
+ '[' 0 -eq 0 ']'
+ printf 'distrobox: Setting up package manager exceptions...\n'
distrobox: Setting up package manager exceptions...
+ '[' -d /usr/lib/rpm/macros.d/ ']'
+ '[' 0 -eq 0 ']'
+ printf 'distrobox: Setting up rpm exceptions...\n'
distrobox: Setting up rpm exceptions...
+ net_mounts=
+ for net_mount in ${HOST_MOUNTS_RO} ${HOST_MOUNTS} '/dev' '/proc' '/sys' '/tmp' '/etc/hosts' '/etc/resolv.conf'
+ net_mounts=/etc/localtime:
+ for net_mount in ${HOST_MOUNTS_RO} ${HOST_MOUNTS} '/dev' '/proc' '/sys' '/tmp' '/etc/hosts' '/etc/resolv.conf'
+ net_mounts=/var/lib/systemd/coredump:/etc/localtime:
+ for net_mount in ${HOST_MOUNTS_RO} ${HOST_MOUNTS} '/dev' '/proc' '/sys' '/tmp' '/etc/hosts' '/etc/resolv.conf'
+ net_mounts=/var/log/journal:/var/lib/systemd/coredump:/etc/localtime:
+ for net_mount in ${HOST_MOUNTS_RO} ${HOST_MOUNTS} '/dev' '/proc' '/sys' '/tmp' '/etc/hosts' '/etc/resolv.conf'
+ net_mounts=/etc/host.conf:/var/log/journal:/var/lib/systemd/coredump:/etc/localtime:
+ for net_mount in ${HOST_MOUNTS_RO} ${HOST_MOUNTS} '/dev' '/proc' '/sys' '/tmp' '/etc/hosts' '/etc/resolv.conf'
+ net_mounts=/etc/machine-id:/etc/host.conf:/var/log/journal:/var/lib/systemd/coredump:/etc/localtime:
+ for net_mount in ${HOST_MOUNTS_RO} ${HOST_MOUNTS} '/dev' '/proc' '/sys' '/tmp' '/etc/hosts' '/etc/resolv.conf'
+ net_mounts=/media:/etc/machine-id:/etc/host.conf:/var/log/journal:/var/lib/systemd/coredump:/etc/localtime:
+ for net_mount in ${HOST_MOUNTS_RO} ${HOST_MOUNTS} '/dev' '/proc' '/sys' '/tmp' '/etc/hosts' '/etc/resolv.conf'
+ net_mounts=/mnt:/media:/etc/machine-id:/etc/host.conf:/var/log/journal:/var/lib/systemd/coredump:/etc/localtime:
+ for net_mount in ${HOST_MOUNTS_RO} ${HOST_MOUNTS} '/dev' '/proc' '/sys' '/tmp' '/etc/hosts' '/etc/resolv.conf'
+ net_mounts=/run/libvirt:/mnt:/media:/etc/machine-id:/etc/host.conf:/var/log/journal:/var/lib/systemd/coredump:/etc/localtime:
+ for net_mount in ${HOST_MOUNTS_RO} ${HOST_MOUNTS} '/dev' '/proc' '/sys' '/tmp' '/etc/hosts' '/etc/resolv.conf'
+ net_mounts=/run/media:/run/libvirt:/mnt:/media:/etc/machine-id:/etc/host.conf:/var/log/journal:/var/lib/systemd/coredump:/etc/localtime:
+ for net_mount in ${HOST_MOUNTS_RO} ${HOST_MOUNTS} '/dev' '/proc' '/sys' '/tmp' '/etc/hosts' '/etc/resolv.conf'
+ net_mounts=/run/netconfig/:/run/media:/run/libvirt:/mnt:/media:/etc/machine-id:/etc/host.conf:/var/log/journal:/var/lib/systemd/coredump:/etc/localtime:
+ for net_mount in ${HOST_MOUNTS_RO} ${HOST_MOUNTS} '/dev' '/proc' '/sys' '/tmp' '/etc/hosts' '/etc/resolv.conf'
+ net_mounts=/run/systemd/journal:/run/netconfig/:/run/media:/run/libvirt:/mnt:/media:/etc/machine-id:/etc/host.conf:/var/log/journal:/var/lib/systemd/coredump:/etc/localtime:
+ for net_mount in ${HOST_MOUNTS_RO} ${HOST_MOUNTS} '/dev' '/proc' '/sys' '/tmp' '/etc/hosts' '/etc/resolv.conf'
+ net_mounts=/run/systemd/resolve/:/run/systemd/journal:/run/netconfig/:/run/media:/run/libvirt:/mnt:/media:/etc/machine-id:/etc/host.conf:/var/log/journal:/var/lib/systemd/coredump:/etc/localtime:
+ for net_mount in ${HOST_MOUNTS_RO} ${HOST_MOUNTS} '/dev' '/proc' '/sys' '/tmp' '/etc/hosts' '/etc/resolv.conf'
+ net_mounts=/run/systemd/seats:/run/systemd/resolve/:/run/systemd/journal:/run/netconfig/:/run/media:/run/libvirt:/mnt:/media:/etc/machine-id:/etc/host.conf:/var/log/journal:/var/lib/systemd/coredump:/etc/localtime:
+ for net_mount in ${HOST_MOUNTS_RO} ${HOST_MOUNTS} '/dev' '/proc' '/sys' '/tmp' '/etc/hosts' '/etc/resolv.conf'
+ net_mounts=/run/systemd/sessions:/run/systemd/seats:/run/systemd/resolve/:/run/systemd/journal:/run/netconfig/:/run/media:/run/libvirt:/mnt:/media:/etc/machine-id:/etc/host.conf:/var/log/journal:/var/lib/systemd/coredump:/etc/localtime:
+ for net_mount in ${HOST_MOUNTS_RO} ${HOST_MOUNTS} '/dev' '/proc' '/sys' '/tmp' '/etc/hosts' '/etc/resolv.conf'
+ net_mounts=/run/systemd/users:/run/systemd/sessions:/run/systemd/seats:/run/systemd/resolve/:/run/systemd/journal:/run/netconfig/:/run/media:/run/libvirt:/mnt:/media:/etc/machine-id:/etc/host.conf:/var/log/journal:/var/lib/systemd/coredump:/etc/localtime:
+ for net_mount in ${HOST_MOUNTS_RO} ${HOST_MOUNTS} '/dev' '/proc' '/sys' '/tmp' '/etc/hosts' '/etc/resolv.conf'
+ net_mounts=/run/udev:/run/systemd/users:/run/systemd/sessions:/run/systemd/seats:/run/systemd/resolve/:/run/systemd/journal:/run/netconfig/:/run/media:/run/libvirt:/mnt:/media:/etc/machine-id:/etc/host.conf:/var/log/journal:/var/lib/systemd/coredump:/etc/localtime:
+ for net_mount in ${HOST_MOUNTS_RO} ${HOST_MOUNTS} '/dev' '/proc' '/sys' '/tmp' '/etc/hosts' '/etc/resolv.conf'
+ net_mounts=/var/lib/libvirt:/run/udev:/run/systemd/users:/run/systemd/sessions:/run/systemd/seats:/run/systemd/resolve/:/run/systemd/journal:/run/netconfig/:/run/media:/run/libvirt:/mnt:/media:/etc/machine-id:/etc/host.conf:/var/log/journal:/var/lib/systemd/coredump:/etc/localtime:
+ for net_mount in ${HOST_MOUNTS_RO} ${HOST_MOUNTS} '/dev' '/proc' '/sys' '/tmp' '/etc/hosts' '/etc/resolv.conf'
+ net_mounts=/var/mnt:/var/lib/libvirt:/run/udev:/run/systemd/users:/run/systemd/sessions:/run/systemd/seats:/run/systemd/resolve/:/run/systemd/journal:/run/netconfig/:/run/media:/run/libvirt:/mnt:/media:/etc/machine-id:/etc/host.conf:/var/log/journal:/var/lib/systemd/coredump:/etc/localtime:
+ for net_mount in ${HOST_MOUNTS_RO} ${HOST_MOUNTS} '/dev' '/proc' '/sys' '/tmp' '/etc/hosts' '/etc/resolv.conf'
+ net_mounts=/dev:/var/mnt:/var/lib/libvirt:/run/udev:/run/systemd/users:/run/systemd/sessions:/run/systemd/seats:/run/systemd/resolve/:/run/systemd/journal:/run/netconfig/:/run/media:/run/libvirt:/mnt:/media:/etc/machine-id:/etc/host.conf:/var/log/journal:/var/lib/systemd/coredump:/etc/localtime:
+ for net_mount in ${HOST_MOUNTS_RO} ${HOST_MOUNTS} '/dev' '/proc' '/sys' '/tmp' '/etc/hosts' '/etc/resolv.conf'
+ net_mounts=/proc:/dev:/var/mnt:/var/lib/libvirt:/run/udev:/run/systemd/users:/run/systemd/sessions:/run/systemd/seats:/run/systemd/resolve/:/run/systemd/journal:/run/netconfig/:/run/media:/run/libvirt:/mnt:/media:/etc/machine-id:/etc/host.conf:/var/log/journal:/var/lib/systemd/coredump:/etc/localtime:
+ for net_mount in ${HOST_MOUNTS_RO} ${HOST_MOUNTS} '/dev' '/proc' '/sys' '/tmp' '/etc/hosts' '/etc/resolv.conf'
+ net_mounts=/sys:/proc:/dev:/var/mnt:/var/lib/libvirt:/run/udev:/run/systemd/users:/run/systemd/sessions:/run/systemd/seats:/run/systemd/resolve/:/run/systemd/journal:/run/netconfig/:/run/media:/run/libvirt:/mnt:/media:/etc/machine-id:/etc/host.conf:/var/log/journal:/var/lib/systemd/coredump:/etc/localtime:
+ for net_mount in ${HOST_MOUNTS_RO} ${HOST_MOUNTS} '/dev' '/proc' '/sys' '/tmp' '/etc/hosts' '/etc/resolv.conf'
+ net_mounts=/tmp:/sys:/proc:/dev:/var/mnt:/var/lib/libvirt:/run/udev:/run/systemd/users:/run/systemd/sessions:/run/systemd/seats:/run/systemd/resolve/:/run/systemd/journal:/run/netconfig/:/run/media:/run/libvirt:/mnt:/media:/etc/machine-id:/etc/host.conf:/var/log/journal:/var/lib/systemd/coredump:/etc/localtime:
+ for net_mount in ${HOST_MOUNTS_RO} ${HOST_MOUNTS} '/dev' '/proc' '/sys' '/tmp' '/etc/hosts' '/etc/resolv.conf'
+ net_mounts=/etc/hosts:/tmp:/sys:/proc:/dev:/var/mnt:/var/lib/libvirt:/run/udev:/run/systemd/users:/run/systemd/sessions:/run/systemd/seats:/run/systemd/resolve/:/run/systemd/journal:/run/netconfig/:/run/media:/run/libvirt:/mnt:/media:/etc/machine-id:/etc/host.conf:/var/log/journal:/var/lib/systemd/coredump:/etc/localtime:
+ for net_mount in ${HOST_MOUNTS_RO} ${HOST_MOUNTS} '/dev' '/proc' '/sys' '/tmp' '/etc/hosts' '/etc/resolv.conf'
+ net_mounts=/etc/resolv.conf:/etc/hosts:/tmp:/sys:/proc:/dev:/var/mnt:/var/lib/libvirt:/run/udev:/run/systemd/users:/run/systemd/sessions:/run/systemd/seats:/run/systemd/resolve/:/run/systemd/journal:/run/netconfig/:/run/media:/run/libvirt:/mnt:/media:/etc/machine-id:/etc/host.conf:/var/log/journal:/var/lib/systemd/coredump:/etc/localtime:
+ net_mounts=/etc/resolv.conf:/etc/hosts:/tmp:/sys:/proc:/dev:/var/mnt:/var/lib/libvirt:/run/udev:/run/systemd/users:/run/systemd/sessions:/run/systemd/seats:/run/systemd/resolve/:/run/systemd/journal:/run/netconfig/:/run/media:/run/libvirt:/mnt:/media:/etc/machine-id:/etc/host.conf:/var/log/journal:/var/lib/systemd/coredump:/etc/localtime
+ cat
+ '[' -d /etc/dpkg/dpkg.cfg.d/ ']'
+ '[' -d /usr/share/libalpm/scripts ']'
+ '[' -d /etc/dpkg/dpkg.cfg.d/ ']'
+ '[' -d /usr/share/libalpm/scripts ']'
+ printf 'distrobox: Setting up distrobox profile...\n'
distrobox: Setting up distrobox profile...
+ touch /run/.toolboxenv
+ mkdir -p /etc/profile.d
+ cat
+ '[' -e /etc/fish/config.fish ']'
+ printf 'distrobox: Setting up sudo...\n'
distrobox: Setting up sudo...
+ mkdir -p /etc/sudoers.d
+ '[' -e /etc/sudoers ']'
+ cat
+ '[' '!' -e /etc/pam.d/su ']'
+ mkdir -p /etc/pam.d
+ cat
+ grep -q pam_systemd.so /etc/pam.d/su
+ '[' 1000 -ne 0 ']'
+ '[' 0 -eq 0 ']'
+ printf '"%s" ALL = (root) NOPASSWD:ALL\n' cdock
+ printf 'distrobox: Setting up user groups...\n'
distrobox: Setting up user groups...
+ grep -q '^cdock:' /etc/group
+ printf 'distrobox: Setting up kerberos integration...\n'
distrobox: Setting up kerberos integration...
+ '[' -d /run/host/var/kerberos ']'
+ printf 'distrobox: Setting up user'\''s group list...\n'
distrobox: Setting up user's group list...
+ additional_groups=
+ grep -q '^sudo' /etc/group
+ grep -q '^wheel' /etc/group
+ additional_groups=wheel
+ '[' 0 -eq 1 ']'
++ printf %s cdock
++ tr '\\' .
+ grep -q '^cdock:' /etc/passwd
+ '[' '!' -e /etc/passwd.done ']'
++ printf %s cdock
++ tr '\\' .
+ grep -q '^cdock:' /etc/passwd
+ printf 'distrobox: Setting up existing user...\n'
distrobox: Setting up existing user...
+ usermod --home /home/cdock --shell /usr/bin/bash --groups wheel --uid 1000 --gid 1000 --login cdock cdock
+ printf 'distrobox: Setting up user home...\n'
distrobox: Setting up user home...
++ getent passwd cdock
++ cut -d: -f6
+ '[' /home/cdock '!=' /home/cdock ']'
+ printf 'distrobox: Ensuring user'\''s access...\n'
distrobox: Ensuring user's access...
+ '[' '!' -e /etc/passwd.done ']'
++ md5sum
++ cut '-d ' -f1
+ temporary_password=3cd45e79196ece79ccfaca1384558fb7
+ printf %s:%s cdock 3cd45e79196ece79ccfaca1384558fb7
+ chpasswd -e
+ printf %s: cdock
+ chpasswd -e
+ '[' 0 -eq 0 ']'
+ passwd_cmd=passwd
+ passwd --help
+ grep -q -- --stdin
+ printf '%s\n%s\n' 3cd45e79196ece79ccfaca1384558fb7 3cd45e79196ece79ccfaca1384558fb7
+ passwd root
New password: Retype new password: passwd: password updated successfully
+ printf %s: root
+ chpasswd -e
+ '[' 0 -eq 1 ']'
+ touch /etc/passwd.done
+ '[' -n '' ']'
+ '[' -n '' ']'
+ '[' 0 -eq 0 ']'
+ printf 'container_setup_done\n'
container_setup_done
+ HOST_WATCH='
		/etc/hosts
		/etc/localtime
		/etc/resolv.conf
	'
++ hostname
+ id=tumbleweed.chungus
+ '[' -e /run/.containerenv ']'
+ . /run/.containerenv
++ engine=podman-4.9.3
++ name=tumbleweed
++ id=85d8324faa2380f31fc625981916473349e6855126c4d6f721aa4fa03431debd
++ image=registry.opensuse.org/opensuse/distrobox:latest
++ imageid=293c6445813faa5e4bffe80ef04e90f6f7142e98fec3dc8104e4591e27ca7d56
++ rootless=1
++ graphRootMounted=1
+ set +x
+ stat /run/host/etc/shadow
++ stat -c %u /run/host/etc/shadow
+ '[' 65534 = 0 ']'
+ '[' -n '' ']'
+ printf 'distrobox: Installing basic packages...\n'
distrobox: Installing basic packages...
++ basename bash
+ shell_pkg=bash
+ '[' bash = ash ']'
+ missing_packages=0
+ dependencies='
	bc
	bzip2
	chpasswd
	curl
	diff
	find
	findmnt
	gpg
	hostname
	less
	lsof
	man
	mount
	passwd
	pigz
	pinentry
	ping
	ps
	rsync
	script
	ssh
	sudo
	time
	tree
	umount
	unzip
	useradd
	wc
	wget
	xauth
	zip
	bash
'
+ for dep in ${dependencies}
+ command -v bc
+ for dep in ${dependencies}
+ command -v bzip2
+ for dep in ${dependencies}
+ command -v chpasswd
+ for dep in ${dependencies}
+ command -v curl
+ for dep in ${dependencies}
+ command -v diff
+ for dep in ${dependencies}
+ command -v find
+ for dep in ${dependencies}
+ command -v findmnt
+ for dep in ${dependencies}
+ command -v gpg
+ for dep in ${dependencies}
+ command -v hostname
+ for dep in ${dependencies}
+ command -v less
+ for dep in ${dependencies}
+ command -v lsof
+ for dep in ${dependencies}
+ command -v man
+ for dep in ${dependencies}
+ command -v mount
+ for dep in ${dependencies}
+ command -v passwd
+ for dep in ${dependencies}
+ command -v pigz
+ for dep in ${dependencies}
+ command -v pinentry
+ for dep in ${dependencies}
+ command -v ping
+ for dep in ${dependencies}
+ command -v ps
+ for dep in ${dependencies}
+ command -v rsync
+ for dep in ${dependencies}
+ command -v script
+ for dep in ${dependencies}
+ command -v ssh
+ for dep in ${dependencies}
+ command -v sudo
+ for dep in ${dependencies}
+ command -v time
+ for dep in ${dependencies}
+ command -v tree
+ for dep in ${dependencies}
+ command -v umount
+ for dep in ${dependencies}
+ command -v unzip
+ for dep in ${dependencies}
+ command -v useradd
+ for dep in ${dependencies}
+ command -v wc
+ for dep in ${dependencies}
+ command -v wget
+ for dep in ${dependencies}
+ command -v xauth
+ for dep in ${dependencies}
+ command -v zip
+ for dep in ${dependencies}
+ command -v bash
+ PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/bin:/sbin:/usr/bin:/usr/sbin
+ '[' 0 -ne 0 ']'
+ '[' 0 -ne 0 ']'
+ '[' -n '' ']'
++ command -v bash
+ SHELL=/usr/bin/bash
+ /usr/bin/distrobox-host-exec -Y test
+ :
+ command -v xdg-open
/usr/local/bin/xdg-open
+ su --help
+ grep -q pty
+ printf 'distrobox: Setting up devpts mounts...\n'
distrobox: Setting up devpts mounts...
+ grep -q tty /etc/group
+ mount -t devpts devpts -o noexec,nosuid,newinstance,ptmxmode=0666,mode=0620,gid=tty /dev/pts/
+ mount --bind /dev/pts/ptmx /dev/ptmx
+ mount --make-rshared /
+ printf 'distrobox: Setting up read-only mounts...\n'
distrobox: Setting up read-only mounts...
+ HOST_MOUNTS_RO='
	/etc/localtime
	/var/lib/systemd/coredump
	/var/log/journal'
+ for host_mount_ro in ${HOST_MOUNTS_RO}
++ get_locked_mount_flags /run/host/etc/localtime
++ source=/run/host/etc/localtime
++ prev=
++ locked_flags=
++ ls /run/host/etc/localtime
++ true
+++ findmnt --noheadings --output OPTIONS --target /run/host/etc/localtime
++ flags=ro,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=270,subvol=/@/.snapshots/4/snapshot
++ '[' -n ro,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=270,subvol=/@/.snapshots/4/snapshot ']'
++ break
++ for flag in nodev noexec nosuid
++ printf %s ro,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=270,subvol=/@/.snapshots/4/snapshot
++ grep -q nodev
++ for flag in nodev noexec nosuid
++ printf %s ro,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=270,subvol=/@/.snapshots/4/snapshot
++ grep -q noexec
++ for flag in nodev noexec nosuid
++ printf %s ro,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=270,subvol=/@/.snapshots/4/snapshot
++ grep -q nosuid
++ printf %s ''
+ locked_flags=
+ mount_bind /run/host/etc/localtime /etc/localtime ro
+ source_dir=/run/host/etc/localtime
+ target_dir=/etc/localtime
+ mount_flags=
+ '[' 3 -gt 2 ']'
+ mount_flags=ro
+ '[' -L /run/host/etc/localtime ']'
++ init_readlink /run/host/etc/localtime
++ ls -l /run/host/etc/localtime
++ grep -Eo '\->.*'
++ cut '-d ' -f2-
++ sed 's|\.\./|/|g'
+ source_dir=/usr/share/zoneinfo/America/New_York
+ printf %s /usr/share/zoneinfo/America/New_York
+ grep -q /run/host
+ source_dir=/run/host/usr/share/zoneinfo/America/New_York
+ '[' '!' -d /run/host/usr/share/zoneinfo/America/New_York ']'
+ '[' '!' -f /run/host/usr/share/zoneinfo/America/New_York ']'
+ '[' -e /etc/localtime ']'
+ findmnt /etc/localtime
+ '[' -L /etc/localtime ']'
+ '[' -d /run/host/usr/share/zoneinfo/America/New_York ']'
+ '[' -f /run/host/usr/share/zoneinfo/America/New_York ']'
++ dirname /etc/localtime
+ '[' '!' -d /etc ']'
+ touch /etc/localtime
+ '[' ro = '' ']'
+ mount --rbind -o ro /run/host/usr/share/zoneinfo/America/New_York /etc/localtime
+ return 0
+ for host_mount_ro in ${HOST_MOUNTS_RO}
++ get_locked_mount_flags /run/host/var/lib/systemd/coredump
++ source=/run/host/var/lib/systemd/coredump
++ prev=
++ locked_flags=
++ ls /run/host/var/lib/systemd/coredump
++ true
+++ findmnt --noheadings --output OPTIONS --target /run/host/var/lib/systemd/coredump
++ flags=rw,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=257,subvol=/@/var
++ '[' -n rw,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=257,subvol=/@/var ']'
++ break
++ for flag in nodev noexec nosuid
++ printf %s rw,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=257,subvol=/@/var
++ grep -q nodev
++ for flag in nodev noexec nosuid
++ printf %s rw,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=257,subvol=/@/var
++ grep -q noexec
++ for flag in nodev noexec nosuid
++ printf %s rw,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=257,subvol=/@/var
++ grep -q nosuid
++ printf %s ''
+ locked_flags=
+ mount_bind /run/host/var/lib/systemd/coredump /var/lib/systemd/coredump ro
+ source_dir=/run/host/var/lib/systemd/coredump
+ target_dir=/var/lib/systemd/coredump
+ mount_flags=
+ '[' 3 -gt 2 ']'
+ mount_flags=ro
+ '[' -L /run/host/var/lib/systemd/coredump ']'
+ '[' '!' -d /run/host/var/lib/systemd/coredump ']'
+ '[' -e /var/lib/systemd/coredump ']'
+ findmnt /var/lib/systemd/coredump
+ '[' -L /var/lib/systemd/coredump ']'
+ '[' -d /run/host/var/lib/systemd/coredump ']'
+ mkdir -p /var/lib/systemd/coredump
+ '[' ro = '' ']'
+ mount --rbind -o ro /run/host/var/lib/systemd/coredump /var/lib/systemd/coredump
+ return 0
+ for host_mount_ro in ${HOST_MOUNTS_RO}
++ get_locked_mount_flags /run/host/var/log/journal
++ source=/run/host/var/log/journal
++ prev=
++ locked_flags=
++ ls /run/host/var/log/journal
++ true
+++ findmnt --noheadings --output OPTIONS --target /run/host/var/log/journal
++ flags=rw,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=257,subvol=/@/var
++ '[' -n rw,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=257,subvol=/@/var ']'
++ break
++ for flag in nodev noexec nosuid
++ printf %s rw,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=257,subvol=/@/var
++ grep -q nodev
++ for flag in nodev noexec nosuid
++ printf %s rw,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=257,subvol=/@/var
++ grep -q noexec
++ for flag in nodev noexec nosuid
++ printf %s rw,relatime,seclabel,ssd,discard=async,space_cache=v2,subvolid=257,subvol=/@/var
++ grep -q nosuid
++ printf %s ''
+ locked_flags=
+ mount_bind /run/host/var/log/journal /var/log/journal ro
+ source_dir=/run/host/var/log/journal
+ target_dir=/var/log/journal
+ mount_flags=
+ '[' 3 -gt 2 ']'
+ mount_flags=ro
+ '[' -L /run/host/var/log/journal ']'
+ '[' '!' -d /run/host/var/log/journal ']'
+ '[' -e /var/log/journal ']'
+ findmnt /var/log/journal
+ umount /var/log/journal
+ '[' -L /var/log/journal ']'
+ '[' -d /run/host/var/log/journal ']'
+ mkdir -p /var/log/journal
+ '[' ro = '' ']'
+ mount --rbind -o ro /run/host/var/log/journal /var/log/journal
+ return 0
+ printf 'distrobox: Setting up read-write mounts...\n'
distrobox: Setting up read-write mounts...
+ '[' -e /var/home/cdock ']'
+ HOST_MOUNTS='
	/etc/host.conf
	/etc/machine-id
	/media
	/mnt
	/run/libvirt
	/run/media
	/run/netconfig/
	/run/systemd/journal
	/run/systemd/resolve/
	/run/systemd/seats
	/run/systemd/sessions
	/run/systemd/users
	/run/udev
	/var/lib/libvirt
	/var/mnt'
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/etc/host.conf /etc/host.conf
+ source_dir=/run/host/etc/host.conf
+ target_dir=/etc/host.conf
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/etc/host.conf ']'
+ '[' '!' -d /run/host/etc/host.conf ']'
+ '[' '!' -f /run/host/etc/host.conf ']'
+ '[' -e /etc/host.conf ']'
+ findmnt /etc/host.conf
+ '[' -L /etc/host.conf ']'
+ '[' -d /run/host/etc/host.conf ']'
+ '[' -f /run/host/etc/host.conf ']'
++ dirname /etc/host.conf
+ '[' '!' -d /etc ']'
+ touch /etc/host.conf
+ '[' '' = '' ']'
+ mount_flags=rslave
+ mount --rbind -o rslave /run/host/etc/host.conf /etc/host.conf
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/etc/machine-id /etc/machine-id
+ source_dir=/run/host/etc/machine-id
+ target_dir=/etc/machine-id
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/etc/machine-id ']'
+ '[' '!' -d /run/host/etc/machine-id ']'
+ '[' '!' -f /run/host/etc/machine-id ']'
+ '[' -e /etc/machine-id ']'
+ findmnt /etc/machine-id
+ '[' -L /etc/machine-id ']'
+ '[' -d /run/host/etc/machine-id ']'
+ '[' -f /run/host/etc/machine-id ']'
++ dirname /etc/machine-id
+ '[' '!' -d /etc ']'
+ touch /etc/machine-id
+ '[' '' = '' ']'
+ mount_flags=rslave
+ mount --rbind -o rslave /run/host/etc/machine-id /etc/machine-id
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/media /media
+ source_dir=/run/host/media
+ target_dir=/media
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/media ']'
+ '[' '!' -d /run/host/media ']'
+ '[' '!' -f /run/host/media ']'
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/mnt /mnt
+ source_dir=/run/host/mnt
+ target_dir=/mnt
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/mnt ']'
+ '[' '!' -d /run/host/mnt ']'
+ '[' -e /mnt ']'
+ findmnt /mnt
+ '[' -L /mnt ']'
+ '[' -d /run/host/mnt ']'
+ mkdir -p /mnt
+ '[' '' = '' ']'
+ mount_flags=rslave
+ mount --rbind -o rslave /run/host/mnt /mnt
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/run/libvirt /run/libvirt
+ source_dir=/run/host/run/libvirt
+ target_dir=/run/libvirt
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/run/libvirt ']'
+ '[' '!' -d /run/host/run/libvirt ']'
+ '[' '!' -f /run/host/run/libvirt ']'
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/run/media /run/media
+ source_dir=/run/host/run/media
+ target_dir=/run/media
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/run/media ']'
+ '[' '!' -d /run/host/run/media ']'
+ '[' -e /run/media ']'
+ findmnt /run/media
+ '[' -L /run/media ']'
+ '[' -d /run/host/run/media ']'
+ mkdir -p /run/media
+ '[' '' = '' ']'
+ mount_flags=rslave
+ mount --rbind -o rslave /run/host/run/media /run/media
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/run/netconfig/ /run/netconfig/
+ source_dir=/run/host/run/netconfig/
+ target_dir=/run/netconfig/
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/run/netconfig/ ']'
+ '[' '!' -d /run/host/run/netconfig/ ']'
+ '[' '!' -f /run/host/run/netconfig/ ']'
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/run/systemd/journal /run/systemd/journal
+ source_dir=/run/host/run/systemd/journal
+ target_dir=/run/systemd/journal
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/run/systemd/journal ']'
+ '[' '!' -d /run/host/run/systemd/journal ']'
+ '[' -e /run/systemd/journal ']'
+ findmnt /run/systemd/journal
+ '[' -L /run/systemd/journal ']'
+ '[' -d /run/host/run/systemd/journal ']'
+ mkdir -p /run/systemd/journal
+ '[' '' = '' ']'
+ mount_flags=rslave
+ mount --rbind -o rslave /run/host/run/systemd/journal /run/systemd/journal
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/run/systemd/resolve/ /run/systemd/resolve/
+ source_dir=/run/host/run/systemd/resolve/
+ target_dir=/run/systemd/resolve/
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/run/systemd/resolve/ ']'
+ '[' '!' -d /run/host/run/systemd/resolve/ ']'
+ '[' '!' -f /run/host/run/systemd/resolve/ ']'
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/run/systemd/seats /run/systemd/seats
+ source_dir=/run/host/run/systemd/seats
+ target_dir=/run/systemd/seats
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/run/systemd/seats ']'
+ '[' '!' -d /run/host/run/systemd/seats ']'
+ '[' -e /run/systemd/seats ']'
+ findmnt /run/systemd/seats
+ '[' -L /run/systemd/seats ']'
+ '[' -d /run/host/run/systemd/seats ']'
+ mkdir -p /run/systemd/seats
+ '[' '' = '' ']'
+ mount_flags=rslave
+ mount --rbind -o rslave /run/host/run/systemd/seats /run/systemd/seats
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/run/systemd/sessions /run/systemd/sessions
+ source_dir=/run/host/run/systemd/sessions
+ target_dir=/run/systemd/sessions
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/run/systemd/sessions ']'
+ '[' '!' -d /run/host/run/systemd/sessions ']'
+ '[' -e /run/systemd/sessions ']'
+ findmnt /run/systemd/sessions
+ '[' -L /run/systemd/sessions ']'
+ '[' -d /run/host/run/systemd/sessions ']'
+ mkdir -p /run/systemd/sessions
+ '[' '' = '' ']'
+ mount_flags=rslave
+ mount --rbind -o rslave /run/host/run/systemd/sessions /run/systemd/sessions
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/run/systemd/users /run/systemd/users
+ source_dir=/run/host/run/systemd/users
+ target_dir=/run/systemd/users
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/run/systemd/users ']'
+ '[' '!' -d /run/host/run/systemd/users ']'
+ '[' -e /run/systemd/users ']'
+ findmnt /run/systemd/users
+ '[' -L /run/systemd/users ']'
+ '[' -d /run/host/run/systemd/users ']'
+ mkdir -p /run/systemd/users
+ '[' '' = '' ']'
+ mount_flags=rslave
+ mount --rbind -o rslave /run/host/run/systemd/users /run/systemd/users
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/run/udev /run/udev
+ source_dir=/run/host/run/udev
+ target_dir=/run/udev
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/run/udev ']'
+ '[' '!' -d /run/host/run/udev ']'
+ '[' -e /run/udev ']'
+ findmnt /run/udev
+ '[' -L /run/udev ']'
+ '[' -d /run/host/run/udev ']'
+ mkdir -p /run/udev
+ '[' '' = '' ']'
+ mount_flags=rslave
+ mount --rbind -o rslave /run/host/run/udev /run/udev
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/var/lib/libvirt /var/lib/libvirt
+ source_dir=/run/host/var/lib/libvirt
+ target_dir=/var/lib/libvirt
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/var/lib/libvirt ']'
+ '[' '!' -d /run/host/var/lib/libvirt ']'
+ '[' '!' -f /run/host/var/lib/libvirt ']'
+ return 0
+ for host_mount in ${HOST_MOUNTS}
+ mount_bind /run/host/var/mnt /var/mnt
+ source_dir=/run/host/var/mnt
+ target_dir=/var/mnt
+ mount_flags=
+ '[' 2 -gt 2 ']'
+ '[' -L /run/host/var/mnt ']'
+ '[' '!' -d /run/host/var/mnt ']'
+ '[' '!' -f /run/host/var/mnt ']'
+ return 0
+ printf 'distrobox: Setting up host'\''s sockets integration...\n'
distrobox: Setting up host's sockets integration...
++ find /run/host/run -path /run/host/run/media -prune -o -path /run/host/run/timeshift -prune -o -name user -prune -o -name bees -prune -o -name nscd -prune -o -name system_bus_socket -prune -o -type s -print
++ :
+ host_sockets='/run/host/run/systemd/inaccessible/sock
/run/host/run/systemd/notify
/run/host/run/systemd/userdb/io.systemd.DynamicUser
/run/host/run/systemd/io.systemd.ManagedOOM
/run/host/run/systemd/journal/dev-log
/run/host/run/systemd/journal/socket
```

</details>

**Desktop:**
 - Are you using podman, docker or lilipod? `podman version 4.9.3`
 - Which version of distrobox? `distrobox: 1.7.0.1`
 - Which host distribution? `openSUSE MicroOS / Aeon`
 - How did you install distrobox? installed by default with distribution.
   Default tumbleweed distrobox created by `distrobox enter`

**Additional context**

I just started trying out Aeon/MicroOS few days ago for the first time. About a day after I installed, Gnome upgraded to version 46. I noticed this problem then. So I reinstalled fresh and it was still there. I *think*, but am not sure, that it *wasn't* present on Gnome 45.

**Comment by starquake** [comment]
I use Arch and updated to Gnome 46 and for me the icons stopped showing too.

**Comment by 89luca89** [comment]
Seems related to Gnome 46 maybe they changed something?

@cdock1029 being on Aeon, can you try rollback to a previous snapshot pre-46 and check if icons work there?

**Comment by cdock1029** [comment]
When I first noticed this I just re-installed the operating system, from a new base .iso that already had 46 on it so I don't have any 45 snapshots.

I don't have experience with Aeon, Gnome, or Distrobox recently as I was on KDE Tumbleweed for past year so wouldn't know if this existed before, but this is a common use case yes? Maybe others recently on Gnome 45 using exports would know. @starquake seems to confirm.

I've been trying to look around and see if any similar issue has been mentioned for other Gnome 46 users outside of distrobox, but maybe is specific to the container/export domain.

**Comment by starquake** [comment]
Yeah I suspect it's a GNOME issue. Seeing that that is the only thing that was updated when the error started occurring. I'll see if I can create bug report for GNOME. I have no idea however about how all this works.

**Comment by cdock1029** [comment]
@89luca89  I had an old Thinkpad that I installed Aeon on before I put it on my main laptop as experiment just to see how it worked and I had tossed it aside and forgot about it. I just checked, it's running Gnome 45 and does not have this issue. I'll report it to Gnome if it hasn't been already.

**Comment by cdock1029** [comment]
It's probably some version of this issue https://github.com/89luca89/distrobox/issues/128

Some discussion on [the Gnome issue page](https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/7531#note_2066704). 

> The only change I'm aware of since 45 is that gnome-shell now no longer falls back to the icon provided by the X window if it could not associate the window with a .desktop file. But that would still mean that an application that was pinned to the dash would show up twice, once for the .desktop file and once for the window.

**Comment by starquake** [comment]
Setting the hostname would be a solution. How do I set the hostname in a distrobox so I can use it with `distrobox assemble create`? Tried `hostname=xxx` but that doesn't seem to work.

**Comment by cdock1029** [comment]
I was able to do this, passing to create worked:

`distrobox create --clone tumbleweed --name tumbleweed-copy --hostname "$(uname -n)"`

**Comment by 89luca89** [comment]
For this I've now reopened the discussion #62 

The idea is to default to host's username for the container, and use $CONTAINER_ID for default prompt

Let's discuss in that issue for the impact of this change on workflows and/or breakages that this could lead

**Comment by oldkentuckymark** [comment]
This also happens in Plasma 6 wayland (Feodra 40). setting the hostname does not fix it either.

**Comment by cdock1029** [comment]
> This also happens in Plasma 6 wayland (Feodra 40). setting the hostname does not fix it either.

@oldkentuckymark 

In addition to the distrobox hostname matching your base system hostname, have you ensured that the StartupWMClass field of the .desktop file matches the actual wmclass the app runs as OR rename the .desktop file to match the running wmclass.

For GNOME you use 'lg' to inspect the window and find the wmclass, details [here](https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/7531#note_2066704)

For Plasma this here describes the method to find the actual wmclass:

https://userbase.kde.org/LatteDock/FAQ#My_launcher_and_its_window_are_not_associated_correctly_OR_my_launcher_is_using_a_low-resolution_icon.3F

https://www.reddit.com/r/kde/comments/msxuj1/if_you_create_your_own_desktop_files_you_can_fix/

**Comment by LyzardKing** [comment]
I have the same problem on a fresh install of ubuntu 24.04

The WMCLASS is correctly set to `Code` in the .desktop file, with xprop reporting 
`WM_CLASS(STRING) = "code", "Code"`

If I change the hostname the icon is picked up correctly, but I can't pin the app to the dock. If I pin from the menu (before starting vscode) it'll then show a duplicate icon on the dock when I start it.

**Comment by cdock1029** [comment]
@LyzardKing Can you confirm this with a different app besides VSCode? Only because Code does something weird by producing 2 different .desktop files that can mess with the icons.

There is one that has wmclass as code-url-handler. In fact sometimes Code itself does run with this wmclass, I think has something to do with Wayland vs x11. In any case, the file that shows the icon in your app list and when pinned should refer by default to the .desktop file with "code" as the wmclass.

If you also have 2 .desktop files for vscode after exorting, move/rename the one names code-url-handler so there aren't any conflicts. Then you may have to log out and/or restart to get a fresh try at it. See if the remaining "code" .desktop file then works, and if the wmclass matches still. I end up (on GNOME) just deleting the 2nd .desktop file and if I have to, I edit the main one and then it works. But with both present it never works correctly for me.

**Comment by LyzardKing** [comment]
@cdock1029 Thanks a lot! Your suggestion works.
I renamed the other .desktop file (named URL Handler, it uses the --open-url option) and now the one existing icon works.

It still needs the hostname to be set to the host one, but with that change it seems to work properly.

This information should probably be automated (on creating the export for vscode) or at least written on the vscode integration man page. If I can get to it I'll open a PR.

**Comment by LyzardKing** [comment]
Just one extra question. Is it possible to change the hostname of an existing distrobox?
With the normal changes to /etc/hostname it isn't persistent

**Comment by cdock1029** [comment]
I think only by cloning into a new box, but it is pretty easy. You could then delete the original and clone again to rename back to "tumbleweed" etc to not have the name be "-copy".


> `distrobox create --clone tumbleweed --name tumbleweed-copy --hostname "$(uname -n)"`

**Comment by LyzardKing** [comment]
Thanks! With that command it then reports the image of the distrobox as the previously existing one instead of the original image (ubuntu:24.04 in this case).
I assume it's not an issue with the image updates in the future, no?

**Comment by rytrix** [comment]
I've tired all of these solutions and none of them work, the WM class is set correctly, hostname is set correctly, the only thing I see is that the app itself shows <untracked> instead of the .desktop name.

**Comment by rytrix** [comment]
> I've tired all of these solutions and none of them work, the WM class is set correctly, hostname is set correctly, the only thing I see is that the app itself shows <untracked> instead of the .desktop name.

Ok, I found a solution but this really seems like a gnome issue. I had to rename the browser to brave-browser.desktop instead of \<box-name\>-brave-browser.desktop. The only other name gnome will track is com.brave.Browser.desktop.

**Comment by tepene** [comment]
> > I've tired all of these solutions and none of them work, the WM class is set correctly, hostname is set correctly, the only thing I see is that the app itself shows <untracked> instead of the .desktop name.
> 
> Ok, I found a solution but this really seems like a gnome issue. I had to rename the browser to brave-browser.desktop instead of <box-name>-brave-browser.desktop. The only other name gnome will track is com.brave.Browser.desktop.

Renaming the desktop file helped in my case as well. Would be nice to have an option in the assemble manifest to specify the `.desktop` file name

**Comment by diogopessoa** [comment]
Hi all,

icons exported from Distrobox are broken when opening in Gnome Dash. Renaming the .desktop file works as a workaround.

My system:

- Fedora 41 Workstation gnome
- GNOME 47.4
- Wayland
- Distrobox 1.8.1.2
